tests/projects/test_remote_pipeline.py

```diff
@@ -0,0 +1,134 @@
+import base64
+import json
+import unittest.mock
+
+import kfp.compiler
+import kubernetes
+import kubernetes.client
+
+import mlrun
+import tests.projects.base_pipeline
+
+
+class TestRemotePipeline(tests.projects.base_pipeline.TestPipeline):
+    def _set_functions(self):
+        self.project.set_function(
+            func=f"{self.assets_path}/remote_pipeline.py",
+            name="func1",
+            image="mlrun/mlrun",
+            handler="func1",
+        )
+        self.project.set_function(
+            func=f"{self.assets_path}/remote_pipeline.py",
+            name="func2",
+            image="mlrun/mlrun",
+            handler="func2",
+        )
+
+    def test_kfp_pipeline_enriched_with_priority_class_name(self, rundb_mock):
+        mlrun.config.config.default_function_priority_class_name = "default-high"
+
+        mlrun.projects.pipeline_context.clear(with_project=True)
+        self.project = self._create_project("remotepipe")
+        self._set_functions()
+        self.project.set_workflow(
+            "p1",
+            workflow_path=str(f'{self.assets_path / "remote_pipeline.py"}'),
+            handler="kfp_pipeline",
+            engine="kfp",
+            local=False,
+        )
+        self.project.save()
+
+        # we are monkey patching kfp.compiler.Compiler._create_workflow,
+        # in the monkey patching we are enriching the pipeline step with the functions priority_class_name
+        # after enriching the compile function passes the enriched workflow to
+        # kfp.compiler.Compiler._write_workflow and that's why we mock _write_workflow to get the
+        # passed args which one of them is workflow
+        kfp.compiler.Compiler._write_workflow = unittest.mock.Mock(return_value=True)
```

Comment:
This test is here to verify that if for example kfp will change the name of the `_create_workflow` method this test will explode
But the way you're building this test is that it is counting on the existence `_write_workflow` which also might be renamed, I suggest to simply let the compile really write to a file, and read this file